modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -421,7 +422,10 @@ void cv::gimpl::GExecutor::reshape(const GMetaArgs& inMetas, const GCompileArgs&
     ade::passes::PassContext ctx{g};
     passes::initMeta(ctx, inMetas);
     passes::inferMeta(ctx, true);
-    m_ops[0].isl_exec->reshape(g, args);
```

Comment:
@rgarnov looking at  #12990 can you please remind why the generic implementation hasn't been done at the time? What were the limitations or complexity we expected from that generic implementation?

Reply:
I believe it was made non-generic just due to the fact that there were no such use cases not to complicate the logic